### PR TITLE
[#2167] refactor: refactor TestGravitinoConnector to enable StaticGuardedByInstance error-prone

### PR DESCRIPTION
**Pull request by coolderli**
### What changes were proposed in this pull request?

`StaticGuardedByInstance` amis to avoid `public static` fields. Because this field is not thread-safe. See more details at https://errorprone.info/bugpattern/StaticGuardedByInstance.

In this MR, I remove some test code in `com.datastrato.gravitino.trino.connector.GravitinoPlugin` and `com.datastrato.gravitino.trino.connector.GravitinoConnectorFactory`.  At the same time, I enable `StaticGuardedByInstance` error-prone.

### Why are the changes needed?

- Fix: #2167 

### Does this PR introduce _any_ user-facing change?

- no

### How was this patch tested?

- original unit tests.

### CODE REVIEW

**Comment by jerryshao** [comment]
@diqiu50 @yuqi1129 please help to review this.

File: trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoConnector.java [hunk 1]
```diff
@@ -70,10 +75,14 @@ protected QueryRunner createQueryRunner() throws Exception {
         properties.put("gravitino.uri", "http://127.0.0.1:8090");
         queryRunner.createCatalog("test1", "gravitino", properties);
       }
-      server.setCatalogConnectorManager(GravitinoPlugin.catalogConnectorManager);
+
+      CatalogConnectorManager catalogConnectorManager =
+          gravitinoPlugin.getCatalogConnectorManager();
```
Comment (diqiu50) [inline trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoConnector.java:80]: I think there is a real Gravitino client in the CatalogConnectorManager, which may be used to load catalogs. You can pass the mocked Gravitino client to the TestGravitinoFactory to initialize the CatalogConnectorManager.

File: trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoPlugin.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    factory.setHook(this::init);
+    return ImmutableList.of(factory);
+  }
+
+  private void init() {
+    catalogConnectorManager = factory.getCatalogConnectorManager();
+  }
+
+  public CatalogConnectorManager getCatalogConnectorManager() {
+    return catalogConnectorManager;
```
Comment (yuqi1129) [inline trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoPlugin.java:28]: Why don't you return `factory.getCatalogConnectorManager()` directly as it also depends on method `GravitinoConnectorFactory#create` to initialize `catalogConnectorManager`;

File: trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoConnector.java (hunk 1 above)
Comment (diqiu50) [reply trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoConnector.java:80]: The code snippet:

`
76 queryRunner.createCatalog("test1", "gravitino", properties);
`
calls the GravitinoConnector.create() function, which instantiates a CatalogConnectorManager. When start() is invoked on the CatalogConnectorManager, it creates an instance of GravitinoClient. This instance is a real GravitinoClient object.

Later, the line:

`
81 catalogConnectorManager.setGravitinoClient(gravitinoClient);
`
replaces the instance with a mocked GravitinoClient.

At this point, the real GravitinoClient may still be performing background tasks.

**Comment by coolderli** [comment]
@diqiu50 @yuqi1129 I have fixed the CI. Please take a look when you have time. Thanks.

File: trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoConnector.java (hunk 1 above)
Comment (coolderli) [reply trino-connector/src/test/java/com/datastrato/gravitino/trino/connector/TestGravitinoConnector.java:79]: Done

**Comment by yuqi1129** [comment]
It's okay with me for the current changes. @diqiu50, do you have any further comments?

**Comment by yuqi1129** [comment]
@coolderli 
Can you resolve the conflicts?

**Review by diqiu50 — DISMISSED** [review]
LGTM

**Review by diqiu50 — APPROVED** [review]

**Pull request merged**